migration/migrating-openshift-3-to-4.adoc

```diff
@@ -0,0 +1,92 @@
+
+You can migrate data to any storage class that is available on the target cluster, for example, from Red Hat Gluster Storage or NFS storage to Red Hat Ceph Storage.
+
+Optionally, you can migrate the {product-title} 3.7 (and later) control plane settings to {product-title} 4.x with the Control Plane Migration Assistant (CPMA). See xref:migration-understanding-cpma_{context}[].
+
+.Prerequisites
+
+* You must have `cluster-admin` privileges.
+* You must have `podman` installed.
+* The source cluster(s) must be {product-title} 3.7 or later.
+* The target cluster must be {product-title} 3.9 or later.
+* The replication repository object storage must support the S3 API, for example, NooBaa, AWS S3, and generic S3 storage providers.
+* If your applications use images from the `openshift` namespace, the required versions of the images must be present on the target cluster. If not, you must link:../openshift_images/image-streams-manage.html#images-imagestreams-update-tag_image-streams-managing[update the `imagestreamtags` references] to use an available version that is compatible with your application.
++
+[NOTE]
+====
+If the `imagestreamtags` cannot be updated, you can manually upload equivalent images to the application namespaces and update the applications to reference them.
+
+The following `imagestreamtags` have been _removed_ from {product-title} 4.2:
+
+* `dotnet:1.0`, `dotnet:1.1`, `dotnet:2.0`
+* `dotnet-runtime:2.0`
+* `mariadb:10.1`
+* `mongodb:2.4`, `mongodb:2.6`
+* `mysql:5.5`, `mysql:5.6`
+* `nginx:1.8`
+* `nodejs:0.10`, `nodejs:4`, `nodejs:6`
+* `perl:5.16`, `perl:5.20`
+* `php:5.5`, `php:5.6`
+* `postgresql:9.2`, `postgresql:9.4`, `postgresql:9.5`
+* `python:3.3`, `python:3.4`
+* `ruby:2.0`, `ruby:2.2`
+====
+
+include::modules/migration-understanding-cam.adoc[leveloffset=+1]
+
+== Installing the CAM Operator
+
+The CAM Operator installed on all clusters:
+
+* {product-title} 3.x: You must install the CAM Operator manually, because OLM is not supported.
+* {product-title} 4.x: You can install the CAM Operator with OLM.
+
+include::modules/migration-installing-migration-operator-manually.adoc[leveloffset=+2]
+include::modules/migration-installing-migration-operator-olm.adoc[leveloffset=+2]
+
+== Configuring cross-origin resource sharing
+
+You enable communication with the CAM by configuring cross-origin resource sharing on {product-title} 3 and 4 (if the {product-title}  4 cluster is not hosting the CAM).
+
+include::modules/migration-configuring-cors-3.adoc[leveloffset=+2]
+
+You can now verify xref:migration-verifying-cors_{context}[cross-origin resource sharing].
```

Comment:
> @apinnick I'm not sure whether you were ready for a peer review, but I heard that this is going to go out this week, and it needed to be peer reviewed and approved after addressing feedback before it could get merged. So I wanted to make sure we had enough time.
> 

Not ready for peer review. Still need some input on Noobaa.

> Let me know if you have any questions on the feedback. Thanks!
> 
> One other item that I couldn't comment directly inline about is that there are darkcircle*.png files, that don't seem to be used. Can you remove these?

They're used in the CR section. I just added them.

Reply:
I put it in so that they wouldn't skip the verification step by accident if they've just configured CORS on OCP 3.

Reply:
I might just work it in as the last step of the procedure. But again, it's fine to leave it as is.

Reply:
Oh, duh! Why didn't I think of that? :-) Probably because I got the verification step so late that it was simpler to stick it in a separate module. I added the step to the configuration procedures (both modules, but we're not going to use the CORS on ocp 4 cluster until next release). Thanks!